audit/audit.go

```diff
@@ -0,0 +1,164 @@
+package audit
+
+import (
+	"context"
+	"github.com/ONSdigital/go-ns/identity"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/pkg/errors"
+	"net/http"
+	"time"
+)
+
+//go:generate moq -out generated_mocks.go -pkg audit . AuditorService OutboundProducer
+
+type contextKey string
+
+const (
+	eventContextKey = contextKey("audit")
+	fromContextKey  = contextKey("from")
+	requestIDHeader = "X-Request-Id"
+	fromHeader      = "from"
```

Comment:
Not entirely sure what the from header is for. It looks like it will be the user who made the request? If so, there is already a `User-Identity` header for this. See here for an example of it being read https://github.com/ONSdigital/go-ns/blob/4001bc2311e9cd2de98a365b6ff5ca6b1065e576/clients/identity/authentication.go#L91